src/redux/sagas/assistantSaga.jsx

```diff
@@ -205,32 +213,102 @@ function* handleSourceCredibilityCall(action) {
   if (action.type === "CLEAN_STATE") return;
 
   try {
+    yield put(
+      setInputSourceCredDetails(
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        true,
+        false,
+        false,
+      ),
+    );
+
     const inputUrl = yield select((state) => state.assistant.inputUrl);
-    const result = yield call(assistantApi.callSourceCredibilityService, [
+    yield take("SET_SCRAPED_DATA"); // wait until linkList has been created
+    const linkList = yield select((state) => state.assistant.linkList);
+    const inputUrlLinkList = [inputUrl].concat(linkList);
+
+    let result = [];
+    let links = [];
+    const batchSize = 20; // batches of links as UDA service has hard limit of 30 seconds
+    const parallelCalls = 2; // parallel calls to service, max two at a time
+    for (let i = 0; i < inputUrlLinkList.length; i += batchSize) {
+      const batchLinks = inputUrlLinkList.slice(i, i + batchSize);
+      const batchLinksString = batchLinks.join(" ");
+      links.push(batchLinksString);
+
+      if (links.length == parallelCalls) {
+        const [batchResult1, batchResult2] = yield all([
+          call(assistantApi.callSourceCredibilityService, [links[0]]),
+          call(assistantApi.callSourceCredibilityService, [links[1]]),
+        ]);
+        links = [];
+
+        if (batchResult1.entities.SourceCredibility) {
+          result = result.concat(batchResult1.entities.SourceCredibility);
+        }
+        if (batchResult2.entities.SourceCredibility) {
+          result = result.concat(batchResult2.entities.SourceCredibility);
+        }
+      }
+    }
```

Comment:
If you have an odd number of batches this will never process the final chunk - needs an exit condition whereby if `links` is not empty when the for loop finishes then it properly handles the final batch.

Reply:
Ah of course, I have added that now.